### PR TITLE
[IMP] mrp, stock: small reception report improvements

### PR DESCRIPTION
**Pull request by ticodoo**
A bunch of small improvements/fixes:

1. Make sales order a hyperlink in report
2. Include products in labels (+ make them stock.move based rather than source based [i.e. clean up code]
3. UX improvements - show allocate button only when something to allocate + make print label button smaller when in mobile
4. Add missing res_model value to action for auto-opening report after picking validation (i.e. fix for barcode version)

Task: 2632884
ENT PR:  https://github.com/odoo/enterprise/pull/20805

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/stock/models/stock_picking.py [hunk 1]
```diff
@@ -1462,7 +1462,9 @@ def action_picking_move_tree(self):
         return action
 
     def action_view_reception_report(self):
-        return self.env["ir.actions.actions"]._for_xml_id("stock.stock_reception_action")
+        action = self.env["ir.actions.actions"]._for_xml_id("stock.stock_reception_action")
+        action['res_model'] = 'report.stock.report_reception'
```
Comment (amoyaux) [inline addons/stock/models/stock_picking.py:1466]: It should be in the xml action no?

Comment (ticodoo) [reply addons/stock/models/stock_picking.py]: Oops! Fixed it

File: addons/stock/report/report_stock_reception.py [hunk 2]
```diff
@@ -86,8 +86,10 @@ def _get_report_values(self, docids, data=None):
         # show potential moves that can be assigned
         for product_id, outs in products_to_outs.items():
             for out in outs:
-                # only show pickings or MOs, no SOs
-                source = out.picking_id if out.picking_id else out._get_source_document()
+                # we expect len(source) = 2 when picking + origin [e.g. SO] and len() = 1 otherwise [e.g. MO]
+                source = (out._get_source_document(),)
+                if out.picking_id and source[0] != out.picking_id:
```
Comment (Whenrow) [inline addons/stock/report/report_stock_reception.py:91]: Test `if not source` before accessing `source[0]`

File: addons/mrp/report/report_stock_reception.xml [hunk 3]
```diff
@@ -1,23 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<odoo>
-
-    <!-- Reception Report Labels -->
-    <template id="report_reception_report_label_mrp">
```
Comment (Whenrow) [inline addons/mrp/report/report_stock_reception.xml:5]: I guess we should ony remove in master to have a migration script. (idem for the ir.action.report)

File: addons/stock/models/stock_picking.py [hunk 4]
```diff
@@ -607,6 +610,36 @@ def _compute_show_validate(self):
...
 
+    @api.depends('state', 'move_lines', 'picking_type_id')
+    def _compute_show_allocation(self):
+        self.show_allocation = False
+        if not self.user_has_groups('stock.group_reception_report'):
+            return
+        for picking in self:
+            picking.show_allocation = picking._get_show_allocation(picking.move_lines, picking.picking_type_id)
+
+    def _get_show_allocation(self, moves, picking_type_id):
```
Comment (Whenrow) [inline addons/stock/models/stock_picking.py:621]: moves can be derived from self.move_lines

**Comment by ticodoo** [comment]
Hello @odoo/rd-runbot @antonylesuisse . Do I qualify for whitelisting? :crossed_fingers:

**Comment by d-fence** [comment]
Exception added and ci/template is green.
Please forward port to master ASAP without applying any change before we remove the exception.

**Comment by Whenrow** [comment]
@d-fence Thanks
robodoo r+ rebase-ff

**Pull request closed without merging**